### ISSUE TITLE
Assimilate warranty coverage table MERCH.ITM2ITM_WARR

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2075]
_Imported from trac ticket 2075.
Created by **sal**
Opened in trac: Jan 17, 2013, 2:12:02 PM
Last modified in trac: Apr 5, 2013, 11:40:26 AM_
***
_by sal:_
This table should be used for:
 1. Related Items section to find warranties
 1. Validation that a product may be covered by a given warranty
 1. Warranty Pricing

```
SQL> select * from itm2itm_warr
  2  where itm_cd = '001503';
 
ITM_CD          WAR_ITM_C WAR_RET_PRC
---------       --------- -----------
001503             W124DW       59.99
001503             W136DW       89.99
001503             W160DW      149.99
```

I imagine `warranties` (or `warrantycoverages`) is a sufficient table name.
___

> From: "Sal Genovese" <sal@ivansmith.net>
> Subject: QUESTION ON WARRANTIES
> Date: January 17, 2013 9:58:00 AM EST
> To: "'Rob Neville'" <rn@retailarchitects.com>
>
> Rob
>  
> We have both warranty skus and treatment skus. For us, a warranty is a 1-1 relationship and is sold on appliances and electronics. Treatment skus are for furniture and can be sold on multiple items on an order. I say that to be sure when using those 2 words we are talking about the same thing. My question is on warranties. We have certain products that are eligible only for certain warranty sku’s. example a dishwasher can only be warranted by 1 of 5 different warranty sku’s.
> 1.       How do I get thoses potential warranty selections to show up under related products? There are some GERS maintenance screens that allow you to link item with warranty sku but in testing one it didn’t then show as a related product in LOFT. See sku number 001503.
> 2.       Can we write some rules to enforce this so that we don’t sell the wrong warranty for an item?
>  
> Thanks
> Sal
>

**Comment by kentbower** [comment]
_**kbower** trac comment on Jan 31, 2013, 6:58:59 AM:_

Sal, please explain how warranty pricing works in relation to the
itm2itm_warr table.

If WAR_RET_PRC is set in that table, then the price is pulled from there
and all other normal pricing functionality is ignored?

Kent

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Feb 1, 2013, 1:41:35 PM:_

Darlene,
We have a question regarding Warranty pricing and want to make sure we have it working correctly for you.  You have the table itm2itm_warr set up with prices like this:
```
SQL> select * from itm2itm_warr where itm_cd = '834725903';

ITM_CD    WAR_ITM_C WAR_RET_PRC
--------- --------- -----------
834725903 940230838       59.99
834725903 940300091       99.99
834725903 940287976      139.99
834725903 940246552      179.99
834725903 940255688      189.99
834725903 940594600      199.99
834725903 940335006      229.99
834725903 940317527      259.99
834725903 940358383      359.99
834725903 940377319      379.99
834725903 940509168      429.99
834725903 940583506      499.99
834725903 940552602      699.99
834725903 940513662      739.99
```

An item can also have Retail and Promotion calendar prices.  If a warranty item has a promotion that could apply, should it over-ride the price in itm2itm_warr?  Could you give us the hierarchy for how the pricing should apply?

Also, we have discussed screen real estate previously, but our plan would be to have the above warranties appear in related items when you have selected 834725903.  That will be a lot to have on the screen, but I'm not sure there is much you can do about it.

Thanks,
Scott

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Mar 28, 2013, 11:32:13 AM:_

_ An item can also have Retail and Promotion calendar prices. If a warranty item has a promotion that could apply, should it over-ride the price in itm2itm_warr?_ **--a promotion price should always pre-order regular retail. Although, in my experience we don't discount the price of warranties.**
_
Could you give us the hierarchy for how the pricing should apply?_**--if a warranty is in a promotion that promotion price should cone first, followed by a retail price calendar price and finally regular retail. Again, we have not discounted warranties at all in the past.**
Thanks,
Darlene

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 28, 2013, 8:45:45 PM:_

We can't seem to get a straight answer from either company.  But I think the answer is: _whichever is the lowest price._

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 28, 2013, 8:55:39 PM:_

```
  CREATE TABLE "MERCH"."ITM2ITM_WARR"
   (    "ITM_CD" VARCHAR2(9) NOT NULL ENABLE,
        "WAR_ITM_CD" VARCHAR2(9) NOT NULL ENABLE,
        "WAR_RET_PRC" NUMBER(13,2) NOT NULL ENABLE,
         CONSTRAINT "ITM2ITM_WARR_PK" PRIMARY KEY ("WAR_ITM_CD", "ITM_CD")
  USING INDEX PCTFREE 10 INITRANS 99 MAXTRANS 255 COMPUTE STATISTICS
  STORAGE(INITIAL 2744320 NEXT 122880 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 22 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "INDEXES"  ENABLE,
         CONSTRAINT "ITMITMWARR_ITM_FKD" FOREIGN KEY ("ITM_CD")
          REFERENCES "INV"."ITM" ("ITM_CD") DISABLE,
         CONSTRAINT "ITMITMWARR_ITM_FKD2" FOREIGN KEY ("WAR_ITM_CD")
          REFERENCES "INV"."ITM" ("ITM_CD") DISABLE
   ) SEGMENT CREATION IMMEDIATE
  PCTFREE 10 PCTUSED 40 INITRANS 99 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 2170880 NEXT 221184 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 22 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "TABLES"

```

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 30, 2013, 8:55:42 AM:_

Ticket [#2256](2256) has been marked as a duplicate of this ticket.

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 30, 2013, 8:57:37 AM:_

(Changeset [[3985]](../commit/8935fd736fae2b10aaf82ed5d1b43fc07122fcd0))
* closes #2075, Assimilate warranty coverage table MERCH.ITM2ITM_WARR (isf, mor)

**Comment by kentbower** [comment]
_**kbower** trac status change on Mar 30, 2013, 8:57:37 AM:_ **closed**

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Apr 4, 2013, 5:20:46 PM:_

In working with electronic warranties, we are not able to use a warranty that is not associated with the related warranties that display in related items.  There are times that a warranty may not be linked in related items or if an item is now on sale, a different warranty based on retail price would be appropriate, but when selected, the error message states that the warrantable items cannot be covered by the warranty.  Is this the way it is supposed to be?  I like the warranties to show up as related items, but we need to have the flexibility to select a different warranty as well.

**Comment by kentbower** [comment]
_**sjohnson** trac status change on Apr 4, 2013, 5:20:46 PM:_ **reopened**

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 5, 2013, 10:06:28 AM:_

Scott, please start including the email header or signature as to who wrote the comment.  Was the above comment Morris or Ivan Smith?

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Apr 5, 2013, 10:27:12 AM:_

This is a Morris go live blocker.

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 5, 2013, 11:40:26 AM:_

(Changeset [[4005]](../commit/2ef93e1338db0feebd968ec92546e1d5274a15e0))
* closes #2075 (Add 'warrantycoverages_enforcement' System Parameter to allow warranty to cover any product or be restricted by warrantycoverages entries) (mor)

**Comment by kentbower** [comment]
_**kbower** trac status change on Apr 5, 2013, 11:40:26 AM:_ **closed**
